### PR TITLE
bpo-35720: Fixing a memory leak in Modules/main.c:pymain_parse_cmdline_impl

### PR DESCRIPTION
**Pull request by Lucas-C**
When the loop in the pymain_read_conf function in this same file
calls pymain_init_cmdline_argv a 2nd time, the pymain->command
buffer of wchar_t is overriden and the previously allocated memory
is never freed.

https://bugs.python.org/issue35720

### CODE REVIEW

**Comment by the-knights-who-say-ni** [comment]
Hello, and thanks for your contribution!

I'm a bot set up to make sure that the project can legally accept your contribution by verifying you have signed the [PSF contributor agreement](https://www.python.org/psf/contrib/contrib-form/) (CLA).

Unfortunately we couldn't find an account corresponding to your GitHub username on [bugs.python.org](https://bugs.python.org/) (b.p.o) to verify you have signed the CLA (this might be simply due to a missing "GitHub Name" entry in your b.p.o account settings). This is necessary for legal reasons before we can look at your contribution. Please follow [the steps outlined in the CPython devguide](https://devguide.python.org/pullrequest/#licensing) to rectify this issue.


You can [check yourself](https://check-python-cla.herokuapp.com/) to see if the CLA has been received.

Thanks again for your contribution, we look forward to reviewing it!

**Comment by Lucas-C** [comment]
The Travis CI build failed with:

    The job exceeded the maximum time limit for jobs, and has been terminated.

The Azure Pipeline build failed because of a column in the `NEWS` entry. I'm removing it:

    Warning, treated as error:
    [whatsnew/changelog:55] ":pymain" found in "Modules/main.c:pymain_parse_cmdline_impl"

File: Modules/main.c [hunk 1]
```diff
@@ -435,7 +435,9 @@ pymain_clear_pymain(_PyMain *pymain)
 {
 #define CLEAR(ATTR) \
     do { \
-        PyMem_RawFree(ATTR); \
+        if (ATTR != NULL) { \
```
Comment (vstinner) [inline Modules/main.c]: The if is useless: PyMem_RawFree(NULL) does nothing, the call is safe.

File: Modules/main.c [hunk 2]
```diff
@@ -1381,7 +1383,8 @@ pymain_read_conf(_PyMain *pymain, _PyCoreConfig *config,
         config->coerce_c_locale = new_coerce_c_locale;
 
         /* The encoding changed: read again the configuration
-           with the new encoding */
+           with the new encoding, but free allocated memory beforehand: */
+        pymain_clear_pymain(pymain);
```
Comment (vstinner) [inline Modules/main.c]: Please move the call after pymain_clear_cmdline() call a few lines before, and leave the comment unchanged.

File: Modules/main.c (hunk 1 above)
Comment (Lucas-C) [reply Modules/main.c]: Ok, removed

File: Modules/main.c (hunk 2 above)
Comment (Lucas-C) [reply Modules/main.c]: Ok, done

File: Misc/NEWS.d/next/Tests/2018-09-09-14-36-59.bpo-34569.okj1Xh.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+The experimental PEP 554 data channels now correctly pass negative PyLong
+objects between subinterpreters on 32-bit systems. Patch by Michael Felt.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Tests/2018-09-09-14-36-59.bpo-34569.okj1Xh.rst]: This change seems to be unrelated!? cc @aixtools

Comment (Lucas-C) [reply Misc/NEWS.d/next/Tests/2018-09-09-14-36-59.bpo-34569.okj1Xh.rst]: Sorry, I included an extra commit by error. This is fixed now
